library/src/androidTest/java/com/google/android/material/motion/runtime/SchedulerTest.java

```diff
@@ -179,6 +341,62 @@ private EndingDelegatedPlan(String text) {
     }
   }
 
+  public class IncrementerTarget {
+    int addCounter = 0;
+    int removeCounter = 0;
+  }
+
+  public static class NamedCounterPlanPerformer extends Performer implements Performer.NamedPlanPerformance {
+
+    @Override
+    public void addPlan(NamedPlan plan, String name) {
+      IncrementerTarget target = getTarget();
+      target.addCounter += 1;
+    }
+
+    @Override
+    public void removePlan(NamedPlan plan, String name) {
+      IncrementerTarget target = getTarget();
+      target.removeCounter += 1;
+    }
+  }
+
+  public static class StoragePlanPerformer extends Performer implements Performer.NamedPlanPerformance {
+
+    @Override
+    public void addPlan(NamedPlan plan, String name) {
+      List<NamedPlan> target = getTarget();
+      target.add(plan);
+    }
+
+    @Override
+    public void removePlan(NamedPlan plan, String name) {
+      List<NamedPlan> target = getTarget();
+      target.add(plan);
+    }
+  }
+
+  public static class GenericPlanPerformer extends Performer implements Performer.NamedPlanPerformance, Performer.PlanPerformance {
+
+    @Override
+    public void addPlan(Plan plan) {
+      TextView target = getTarget();
+      target.setText(target.getText() + "regularAddPlanInvoked");
+    }
+
+    @Override
+    public void addPlan(NamedPlan plan, String name) {
+      TextView target = getTarget();
+      target.setText(target.getText() + "addPlanInvoked");
+    }
+
+    @Override
+    public void removePlan(NamedPlan plan, String name) {
+      TextView target = getTarget();
+      target.setText(target.getText() + "removePlanInvoked");
```

Comment:
pretty clever way to test 👍 

I would add a space in front like `" removePlanInvoked"` so that the assertions are easier to read: 

```
assertTrue(textView.getText().equals(" removePlanInvoked addPlanInvoked"));
```